site/docs/cow-and-mor.md

```diff
@@ -0,0 +1,195 @@
+<!--
+ - Licensed to the Apache Software Foundation (ASF) under one or more
+ - contributor license agreements.  See the NOTICE file distributed with
+ - this work for additional information regarding copyright ownership.
+ - The ASF licenses this file to You under the Apache License, Version 2.0
+ - (the "License"); you may not use this file except in compliance with
+ - the License.  You may obtain a copy of the License at
+ -
+ -   http://www.apache.org/licenses/LICENSE-2.0
+ -
+ - Unless required by applicable law or agreed to in writing, software
+ - distributed under the License is distributed on an "AS IS" BASIS,
+ - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ - See the License for the specific language governing permissions and
+ - limitations under the License.
+ -->
+
+# Copy-on-Write and Merge-on-Read
+
+This page explains the concept of copy-on-write and merge-on-read in the context of Iceberg to provide readers more clarity around Iceberg's table spec design.
+
+## Introduction
+
+In Iceberg, copy-on-write and merge-on-read are different ways to handle row-level update and delete operations. Here are their definitions:
+
+- **copy-on-write (CoW)**: an update/delete directly rewrites the entire affected data files.
+- **merge-on-read (MoR)**: update/delete information is encoded in the form of delete files. The table reader can apply all delete information at read time. A compaction process takes care of merging delete files into data files asynchronously. 
```

Comment:
I think that this is really hard to read using `CoW` and `MoR` in place of copy-on-write and merge-on-read. I think that we should not shorten them, but I'd like to hear if other people share this opinion.

Reply:
yeah, I am always not fan of abbreviation, they sometimes tend to add more confusion

Reply:
+1, and the articles are tricky as well depending on how you read the abbreviations. For example "a MoR" vs. "an MoR", which one sounds right depends on if you read it as the acronym or expand it to the full "merge-on-read" as you're reading.